homeassistant/components/aftership/sensor.py

```diff
@@ -56,7 +75,40 @@
                       aftership.meta)
         return
 
-    async_add_entities([AfterShipSensor(aftership, name)], True)
+    instance = AfterShipSensor(aftership, name)
+
+    async_add_entities([instance], True)
+
+    async def handle_add_tracking(call):
+        """Call when a user adds a new Aftership tracking from HASS."""
+        title = call.data.get(CONF_TITLE)
+        slug = call.data.get(CONF_SLUG)
+        tracking_number = call.data[CONF_TRACKING_NUMBER]
+
+        await aftership.add_package_tracking(tracking_number, title, slug)
+        await instance.async_update(no_throttle=True)
```

Comment:
Technically it's possible to set up multiple platforms which would make the service handler unreliable.

Should we try to handle that case?

Reply:
It would be good as long as I'm here, but I'm afraid I don't see where the problem would arise. Any guidance would be appreciated. Thanks.

Reply:
When registering a service, any previously registered service handler for the same name will be overwritten. So the reference to the previously added entity will be gone, if setting up a second, etc, platform. So we'll have to store all entities in a list on hass.data and use those in the service handlers.

Reply:
Another way is to use our dispatch helper and connect the entity update method to a signal name in `async_added_to_hass`. Then send that signal here instead of accessing the entity directly. Then we don't need to keep track of the entities.

Reply:
Very slick! Thanks.

Reply:
See below how to change here.